frontend/frontend-filters-links.php

```diff
@@ -445,4 +449,38 @@ public function check_canonical_url( $requested_url = '', $do_redirect = true )
 
 		return $redirect_url;
 	}
+
+	/**
+	 * Returns the term_id of the requested term.
+	 *
+	 * @since 2.8.4
+	 *
+	 * @param string $taxonomy     Taxonomy name
+	 * @param array  $queried_term {
+	 *   Information about the queried term
+	 *   @type string $field The field used to query the term(s).
+	 *   @type mixed  $term  Taxonomy term(s).
+	 * }
+	 * @return int
+	 */
+	protected function get_queried_term_id( $taxonomy, $queried_term ) {
+		$field = $queried_term['field'];
+		$term  = reset( $queried_term['terms'] );
+
+		// We can get a term_id when requesting a plain permalink, eg /?cat=1.
+		if ( 'term_id' === $field ) {
+			return $term;
+		}
```

Comment:
You shouldn't need this if you follow the suggestions from the previous comment.